modules/yugabytedb/yugabytedb_test.go

```diff
@@ -0,0 +1,110 @@
+package yugabytedb_test
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/yugabyte/gocql"
+
+	_ "github.com/lib/pq"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/modules/yugabytedb"
+)
+
+func TestYugabyteDB_YSQL(t *testing.T) {
+	t.Run("Run", func(t *testing.T) {
+		ctx := context.Background()
+
+		ctr, err := yugabytedb.Run(ctx, "yugabytedb/yugabyte:2024.1.3.0-b105")
+		testcontainers.CleanupContainer(t, ctr)
+		require.NoError(t, err)
+
+		ctrHost, err := ctr.Host(ctx)
+		require.NoError(t, err)
+
+		ctrPort, err := ctr.MappedPort(ctx, "5433/tcp")
+		require.NoError(t, err)
+
+		ysqlConnStr, err := ctr.YSQLConnectionString(ctx, "sslmode=disable")
+		require.NoError(t, err)
+		assert.Equal(t, fmt.Sprintf("postgres://yugabyte:yugabyte@%s:%s/yugabyte?sslmode=disable", ctrHost, ctrPort.Port()), ysqlConnStr)
+
+		db, err := sql.Open("postgres", ysqlConnStr)
+		require.NoError(t, err)
+		require.NotNil(t, db)
+
+		err = db.Ping()
+		require.NoError(t, err)
+	})
+
+	t.Run("Run with custom options", func(t *testing.T) {
+		ctx := context.Background()
+		opts := []testcontainers.ContainerCustomizer{
+			yugabytedb.WithYSQLDatabaseName("custom-db"),
+			yugabytedb.WithYSQLDatabaseUser("custom-user"),
+			yugabytedb.WithYSQLDatabasePassword("custom-password"),
+		}
+
+		ctr, err := yugabytedb.Run(ctx, "yugabytedb/yugabyte:2024.1.3.0-b105", opts...)
+		testcontainers.CleanupContainer(t, ctr)
+		require.NoError(t, err)
+
+		ctrHost, err := ctr.Host(ctx)
+		require.NoError(t, err)
+
+		ctrPort, err := ctr.MappedPort(ctx, "5433/tcp")
+		require.NoError(t, err)
+
+		ysqlConnStr, err := ctr.YSQLConnectionString(ctx, "sslmode=disable")
+		require.NoError(t, err)
+		assert.Equal(t, fmt.Sprintf("postgres://custom-user:custom-password@%s:%s/custom-db?sslmode=disable", ctrHost, ctrPort.Port()), ysqlConnStr)
+
+		db, err := sql.Open("postgres", ysqlConnStr)
+		require.NoError(t, err)
+		require.NotNil(t, db)
+
+		err = db.Ping()
+		require.NoError(t, err)
+	})
+}
+
+func TestYugabyteDB_YCQL(t *testing.T) {
+	t.Run("Run", func(t *testing.T) {
+		ctx := context.Background()
+
+		ctr, err := yugabytedb.Run(ctx, "yugabytedb/yugabyte:2024.1.3.0-b105")
+		testcontainers.CleanupContainer(t, ctr)
+		require.NoError(t, err)
+
+		cluster := gocql.NewCluster()
+		ctr.YCQLConfigureClusterConfig(ctx, cluster)
+
+		session, err := cluster.CreateSession()
+		require.NoError(t, err)
+		t.Cleanup(func() { session.Close() })
```

Comment:
suggestion: call directly and check error:
```suggestion
		require.NoError(t, session.Close())
```

Reply:
I think I saw this fixed in a previous version, did it get lost? Another below too, as we only need a t.Cleanup if the test is going to continue.

Reply:
I've reverted it as the `Close` method of the `gocpl.Session` does not return an error